onmt/inputters/inputter.py

```diff
@@ -167,9 +187,9 @@ def dict_to_vocabs(vocabs_dict):
     else:
         vocabs['tgt'] = pyonmttok.build_vocab_from_tokens(vocabs_dict['tgt'])
     if 'src_feats' in vocabs_dict.keys():
-        vocabs['src_feats'] = {}
-        for feat in vocabs_dict['src_feats'].keys():
-            vocabs['src_feats'][feat] = \
+        vocabs['src_feats'] = []
+        for feat_vocab in vocabs_dict['src_feats']:
+            vocabs['src_feats'].append(
                 pyonmttok.build_vocab_from_tokens(
-                    vocabs_dict['src_feats'][feat])
+                    feat_vocab))
     return vocabs
```

Comment:
don't we need to add a section for vocabs['tgt_feats'] as well ?

Reply:
Sure! Changed in the next commit